src/example_provision.rs

```diff
@@ -66,72 +71,221 @@ enum ExampleError {
+    registry.register(
+        ActionName::new("instance_create_params"),
+        new_action_noop_undo(demo_prov_instance_create_params),
+    );
+
+    registry.register(
+        ActionName::new("instance_create"),
         new_action_noop_undo(demo_prov_instance_create),
     );
-    w.append_parallel(vec![
-        (
+    registry.register(
+        ActionName::new("vpc_alloc_ip"),
+        new_action_noop_undo(demo_prov_vpc_alloc_ip),
+    );
+    registry.register(
+        ActionName::new("volume_create"),
+        new_action_noop_undo(demo_prov_volume_create),
+    );
+    registry.register(
+        ActionName::new("instance_configure"),
+        new_action_noop_undo(demo_prov_instance_configure),
+    );
+    registry.register(
+        ActionName::new("volume_attach"),
+        new_action_noop_undo(demo_prov_volume_attach),
+    );
+    registry.register(
+        ActionName::new("instance_boot"),
+        new_action_noop_undo(demo_prov_instance_boot),
+    );
+    registry.register(
+        ActionName::new("print"),
+        new_action_noop_undo(demo_prov_print),
+    );
+
+    // Subsaga actions are registered just like any other action
+    // The order of registration doesn't matter
+    registry.register(
+        ActionName::new("server_alloc_params"),
+        new_action_noop_undo(demo_prov_server_alloc_params),
+    );
+    registry.register(
+        ActionName::new("server_pick"),
+        new_action_noop_undo(demo_prov_server_pick),
+    );
+    registry.register(
+        ActionName::new("server_reserve"),
+        new_action_noop_undo(demo_prov_server_reserve),
+    );
+
+    Arc::new(registry)
+}
+
+// Create a subsaga for server allocation
+pub fn server_alloc_subsaga<'a>() -> SagaSpec<'a> {
+    SagaSpec(
+        NodeSpec {
+            name: "server_alloc_params",
+            label: "ReturnServerAllocParams",
+            action: "server_alloc_params",
+        },
+        vec![NodeConcurrency::Linear(vec![
+            NodeSpec {
+                name: "server_id",
+                label: "ServerPick",
+                action: "server_pick",
+            },
+            NodeSpec {
+                name: "server_reserve",
+                label: "ServerReserve",
+                action: "server_reserve",
+            },
+        ])],
+    )
+}
+
+/// Create a dag that describes a "VM Provision" Saga
+///
+/// The actions in this saga do essentially nothing. They print out what
+/// node is running, they produce some data, and they consume some data
+/// from previous nodes. The intent is just to exercise the API. You can
+/// interact with this  using the `demo-provision` example.
+pub fn make_example_provision_dag(params: &ExampleParams) -> Arc<Dag> {
+    let name = SagaName::new("DemoVmProvision");
+    let mut d = DagBuilder::new(name);
+
+    // The saga instance Id (not related to VM instance)
+    // TODO(AJS): Name this something else?
+    let saga_instance_id = 0;
+    d.append(Node::new_root(
```

Comment:
Note that this whole saga could either be created via `SagaSpec`(s) or manually through builder calls to `append` and `append_parallel`.